src/core/client/local-tool-provider.ts

```
@@ -0,0 +1,216 @@
+    async readResource(): Promise<any> {
+        throw new Error('Resources not supported by LocalToolProvider');
+    }
+
+    async getConnectedClient(): Promise<Client> {
+        throw new Error('LocalToolProvider has no underlying MCP client');
+    }
+
+    /**
+     * Extract description from function comments or generate a reasonable default
+     */
+    private extractDescription(fn: Function, fnName: string): string {
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)